src/object_accessor.hpp

```diff
@@ -246,23 +250,38 @@ Object Object::create(ContextType ctx, SharedRealm realm, const ObjectSchema &ob
         row_index = get_for_primary_key_impl(ctx, *table, *primary_prop, primary_value);
 
         if (row_index == realm::not_found) {
-            row_index = table->add_empty_row();
             created = true;
-            if (primary_prop->type == PropertyType::Int)
+            if (primary_prop->type == PropertyType::Int) {
+#if REALM_ENABLE_SYNC
+                row_index = sync::create_object_with_primary_key(realm->read_group(), *table, Accessor::to_long(ctx, primary_value));
```

Comment:
Good question — I should document this!

`sync::create_object_with_primary_key()` and `sync::create_object()` should work on non-sync'ed Realms. They check dynamically if the Realm is sync'ed or not. :)